src/spxmod/dimension.py

```diff
@@ -14,8 +14,9 @@ class Dimension:
 
     """
 
-    def __init__(self, name: str) -> None:
+    def __init__(self, name: str, filter: str | None = None) -> None:
```

Comment:
Thanks very much @kels271828! This is really cool, and it is very flexible. I am a little worry that it will depend on `data` contains all the necessary columns in `query`, and before it will only require `data` has `self.name`. Spin from this, maybe we can let user instead of providing `filter`, we can let them provide a reference value. And we remove `ref` (if any) from the `self.span`. But this will require to set the `location_id` to `-1` when `national_id == location_id`. Just an idea, let me know what you think :)